test/auto_parallel/spmd_rules/test_reduction_rule.py

```diff
@@ -233,6 +235,171 @@ def test_multi_mesh_dim(self):
         self.assertEqual(infered_output_dist_attrs[0]._is_partial(), True)
         self.assertEqual(infered_output_dist_attrs[0]._partial_dims(), {1})
 
+    def test_backward_single_mesh_dim(self):
+        # reduce on dim 0, keep_dim = false
+        # [-1] --> [-1, -1], [-1] (output --> input, output)
+        self.attrs['keep_dim'] = False
+        self.attrs['axis'] = [0]
+        self.out_dist_tensor_spec.shape = [32]
+        self.out_dist_tensor_spec.set_dims_mapping([-1])
+        result_dist_attrs = self.rule.infer_backward(
+            [self.x_dist_tensor_spec], [self.out_dist_tensor_spec], self.attrs
+        )
+        infered_input_dist_attrs = result_dist_attrs[0]
+        infered_output_dist_attrs = result_dist_attrs[1]
+
+        self.assertEqual(len(result_dist_attrs), 2)
+        self.assertEqual(len(infered_input_dist_attrs), 1)
+        self.assertEqual(len(infered_output_dist_attrs), 1)
+
+        self.assertEqual(infered_input_dist_attrs[0].dims_mapping, [-1, -1])
+        self.assertEqual(infered_output_dist_attrs[0].dims_mapping, [-1])
+
+        # reduce on dim 0, keep_dim = true
```

Comment:
should add a kind new case that output is partial on curtain dims, and how it influence input :
stage 1: input is not effect
stage2: input should be sharded on that dims